rlog/2025-06-25-gsub-perf-imp-comparison.mdx

```diff
@@ -0,0 +1,392 @@
+
+Rotating publishers ensures that every message traverses a different path, 
+which helps achieve fair performance evaluation. 
+On the other hand, changing inter-message delays allows for creating varying traffic patterns. 
+A shorter inter-message delay implies more messages can be in transit simultaneously, 
+which helps evaluate performance against large message counts. 
+A longer delay ensures every message is fully disseminated before introducing a new message. 
+Similarly, increasing message size stresses the network. 
+As a result, we evaluate performance across a broader range of use cases. 
+
+
+The simulation details are presented in the tables below. 
+The experiments are conducted using the shadow simulator. 
+We uniformly set peer bandwidths and link latencies between 40-200 Mbps and 
+40-130 milliseconds in five variations. 
+
+
+**Table 1:** Simulation Scenarios.
+| **Experiment** | **No. of Nodes** | **No. of Publishers** | **Message Size (KB)** | **Inter-Message Delay (ms)** |
+|:-------------:|:-------------:|:---------:|:-------------:|:---------:|
+| Scenario 1 | 3000, 6000, 9000, 12000 | 7 | 150 | 10000 |
+| Scenario 2 | 1500 | 10 | 200, 600, 1000, 1400, 1800 | 10000 |
+| Scenario 3 | 1500 | 25, 50, 75, 100, 125 | 50 | 50 |
+
+
+**Table 2:** Simulation Parameters.
+| **Parameter** | **Value** | **Parameter** | **Value** |
+|:-------------:|:---------:|:-------------:|:---------:|
+| $D$ | 8 | $D_{low}$ | 6 |
+| $D_{lazy}$ | 6 | $D_{high}$ | 12 |
+| Gossip factor | 0.05 | Muxer | yamux |
+| Heartbeat interval | 1000 ms | Floodpublish | false |
+| Peer Bandwidth | 40-200 Mbps | Link Latency| 40-130ms |
+| $D_{announce}$ GossipSub v2.0| 7 | Forward delay in PPPT/v1.4 with delay| 35 ms |
+
+
+### Results
+
+#### Scenario1: Increasing Network Size
+
+| ![Bandwidth](/img/gsub-imp-comparison/S1_BW.png) | ![Latency](/img/gsub-imp-comparison/S1_Lat.png) | ![Duplicates](/img/gsub-imp-comparison/S1_Dup.png) | ![IWANTs](/img/gsub-imp-comparison/S1_IWANT.png) |
+|:--:|:--:|:--:|:--:|
+| Bandwidth | Latency | Average Duplicates | Average IWANT Requests |
+
+
+#### Scenario2: Increasing Message Size
+
+| ![Bandwidth](/img/gsub-imp-comparison/S2_BW.png) | ![Latency](/img/gsub-imp-comparison/S2_Lat.png) | ![Duplicates](/img/gsub-imp-comparison/S2_Dup.png) | ![IWANTs](/img/gsub-imp-comparison/S2_IWANT.png) |
+|:--:|:--:|:--:|:--:|
+| Bandwidth | Latency | Average Duplicates | Average IWANT Requests |
+
+
+#### Scenario3: Increasing Number of Publishers
+
+| ![Bandwidth](/img/gsub-imp-comparison/S3_BW.png) | ![Latency](/img/gsub-imp-comparison/S3_Lat.png) | ![Duplicates](/img/gsub-imp-comparison/S3_Dup.png) | ![IWANTs](/img/gsub-imp-comparison/S3_IWANT.png) |
+|:--:|:--:|:--:|:--:|
+| Bandwidth | Latency | Average Duplicates | Average IWANT Requests |
+
+
+Please feel free to join the discussion and leave feedback regarding this post in the

```

Comment:
Would it make sense to add a short “conclusion” here with 2–3 concrete takeaways? even stating what approach you consider should be included in libp2p, or what needs more testing? the call to action to join the discussion / leaving feedback could be part of this conclusion section